hummingbot/strategy/avellaneda_market_making/avellaneda_market_making_config_map_pydantic.py

```diff
@@ -0,0 +1,529 @@
+        ge=1,
+        le=10_000,
+        client_data=ClientFieldData(
+            prompt=lambda mi: "Enter amount of ticks that will be stored to estimate order book liquidity",
+        ),
+    )
+    order_levels_mode: Union[MultiOrderLevelModel, SingleOrderLevelModel] = Field(
+        default=SingleOrderLevelModel.construct(),
+        description="Allows activating multi-order levels.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: f"Select the order levels mode ({'/'.join(OrderLevelsMode)}",
+        ),
+    )
+    order_override: Optional[Dict] = Field(
+        default=None,
+        description="Allows custom specification of the order levels and their spreads and amounts.",
+        client_data=None,
+    )
+    hanging_orders_mode: Union[TrackHangingOrdersModel, IgnoreHangingOrdersModel] = Field(
+        default=IgnoreHangingOrdersModel.construct(),
+        description="When tracking hanging orders, the orders on the side opposite to the filled orders remain active.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: f"How do you want to handle hanging orders? ({'/'.join(HangingOrdersMode)})",
+        ),
+    )
+    should_wait_order_cancel_confirmation: bool = Field(
+        default=True,
+        description=(
+            "If activated, the strategy will await cancellation confirmation from the exchange"
+            " before placing a new order."
+        ),
+        client_data=ClientFieldData(
+            prompt=lambda mi: (
+                "Should the strategy wait to receive a confirmation for orders cancellation"
+                " before creating a new set of orders?"
+                " (Not waiting requires enough available balance) (Yes/No)"
+            ),
+        )
+    )
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        required_exchanges.append(self.exchange)
+
+    class Config:
+        validate_assignment = True
+
+    @validator("exchange", pre=True)
+    def validate_exchange(cls, v: str):
+        ret = validate_exchange(v)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+    @validator("market", pre=True)
+    def validate_exchange_trading_pair(cls, v: str, values: Dict):
+        exchange = values.get("exchange")
+        ret = validate_market_trading_pair(exchange, v)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+    @validator("execution_timeframe_model", pre=True)
+    def validate_execution_timeframe(
+        cls, v: Union[str, InfiniteModel, FromDateToDateModel, DailyBetweenTimesModel]
+    ):
+        if not isinstance(v, Dict) and not hasattr(ExecutionTimeframe, v):
+            raise ValueError(
+                f"Invalid timeframe, please choose value from {[e.value for e in list(ExecutionTimeframe)]}"
+            )
+        elif isinstance(v, (InfiniteModel, FromDateToDateModel, DailyBetweenTimesModel)):
+            sub_model = v
```

Comment:
Is this case Possible? Because in this case v is not a Dict, and ExecutionTimeFrame does not have instances of this classes as attributes, then the previous if statement should be True and raise an error.

Reply:
Yes, good point. I wanted to add this check in case a sub-model was already provided as the value here, but I don't currently have a scenario where that happens... I will still keep the check for completeness, but will move it to the top